src/CascadeSoftDeletes.php

```diff
@@ -33,8 +33,12 @@ protected static function bootCascadeSoftDeletes()
                 ));
             }
 
+            $delete = $model->forceDeleting ? 'forceDelete' : 'delete';
+
             foreach ($model->getCascadingDeletes() as $relationship) {
-                $model->forceDeleting ? $model->{$relationship}()->forceDelete() : $model->{$relationship}()->delete();
+                foreach ($model->{$relationship} as $child) {
```

Comment:
`$model->{$relationship} as $child` forces Eloquent to create `$child` model instances, which in turn trigger delete events.